### PR TITLE
[pytorch][mobile] remove backward functions from jit-op-registry for mobile build

### PR DESCRIPTION
**Pull request by ljk53**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26783 [pytorch][mobile] remove backward functions from jit-op-registry for mobile build**
* #26702 [pytorch][perf] add mobile friendly at:parallel_for backend

Summary:
Add codegen option to remove backward ops from jit-op-registry as they are not
likely to be used for inference only mobile build.

Measured ARM-v7 AAR build size change: 5,804,182 -> 5,331,219.

Test Plan:
- build and integrate with demo app;

Differential Revision: [D17570054](https://our.internmc.facebook.com/intern/diff/D17570054)

### CODE REVIEW

**Comment by ljk53** [comment]
Recreate #26657 - that PR was marked as merged for some reason.

File: tools/jit/gen_jit_dispatch.py [hunk 1]
```diff
@@ -238,6 +239,19 @@ def is_out_variant(decl):
     return decl['name'].endswith('_out')
 
 
+# Copied from ..autograd.gen_python_functions.SKIP_PYTHON_BINDINGS
```
Comment (ljk53) [inline tools/jit/gen_jit_dispatch.py]: @zdevito is there a reason to bind other skipped python bindings for JIT? Can we safely skip all of them by default?

Comment (dzhulgakov) [reply tools/jit/gen_jit_dispatch.py]: There's symbolic differentiation support in JIT. It'd append IR for backwards directly to the graph. For those cases we need to have backwards ops registered and runnable. (at least that's my understanding)

**Review by dzhulgakov — COMMENTED** [review]
Looks good, but I'll let @ezyang or @zdevito  to take a look

**Comment by ljk53** [comment]
ahh... this PR got marked as merged again.. I'll recreate.

**Pull request merged**